pkg/network/setup/netstat.go

```diff
@@ -131,10 +131,19 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
 	}
 
-	for ifaceIndex, ifaceStatus := range interfacesStatus {
-		if _, exists := multusStatusNetworksByName[ifaceStatus.Name]; exists {
-			interfacesStatus[ifaceIndex].InfoSource = netvmispec.AddInfoSource(
-				ifaceStatus.InfoSource, netvmispec.InfoSourceMultusStatus)
+	indexedIfaceStatus := netvmispec.IndexInterfacesFromStatus(interfacesStatus,
+		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool { return true })
+	for multusIfaceName := range multusStatusNetworksByName {
+		_, existInSpec := vmiInterfacesSpecByName[multusIfaceName]
+		_, existInNewStatus := indexedIfaceStatus[multusIfaceName]
+		if existInSpec && !existInNewStatus {
+			interfacesStatus = append(interfacesStatus, v1.VirtualMachineInstanceNetworkInterface{
+				Name:       multusIfaceName,
+				InfoSource: netvmispec.InfoSourceMultusStatus,
+			})
+		} else if existInNewStatus {
+			iface := netvmispec.LookupInterfaceStatusByName(interfacesStatus, multusIfaceName)
```

Comment:
This is the only lookup needed, the `indexedIfaceStatus` is not needed, it is a duplication AFAIU.

Reply:
This lookup is performed to mutate the actual interfaces that will be set on the VMI object.
Mutating `indexedIfaceStatus` iteration objected won't  because it's not the same underlying object

Reply:
I think the comment was about not needing the indexing and just doing this lookup. Checking if it exists or not, if it exists it is updated, if it does not it is added.

Reply:
I am writing the whole suggestion here to be clear.
Note, this whole block needs to come in a function like all the other source collections above.

```go
func ifacesStatusFromMultus(interfacesStatus []v1.VirtualMachineInstanceNetworkInterface, multusStatusNetworksByName map[string]blabla, vmIfacesSpecByName map[string]blabla) []v1.VirtualMachineInstanceNetworkInterface {
	for ifaceName := range multusStatusNetworksByName {
	        ifaceStatus := netvmispec.LookupInterfaceStatusByName(interfacesStatus, ifaceName)
                _, existInSpec := vmIfacesSpecByName[ifaceName]
                if existInSpec && ifaceStatus == nil {
			interfacesStatus = append(interfacesStatus, v1.VirtualMachineInstanceNetworkInterface{
				Name:       ifaceName,
				InfoSource: netvmispec.InfoSourceMultusStatus,
			})
		} else if ifaceStatus != nil {
			ifaceStatus.InfoSource = netvmispec.AddInfoSource(ifaceStatus.InfoSource, netvmispec.InfoSourceMultusStatus)
		}
	}
        return interfacesStatus
}
```

Reply:
Thanks for the example, done